lib/puppet/provider/scheduled_task/taskscheduler_api2.rb

```diff
@@ -207,4 +207,10 @@ def validate_trigger(value)
 
     true
   end
+
+  def validate_name
+    if @resource[:name].match?(/\\/) && @resource[:compatibility] < 2
+      raise Puppet::ResourceError, "#{@resource[:name]} specifies a path including subfolders and a compatibility of #{@resource[:compatibility]} - tasks in subfolders are only supported version 2 and later of the API. Specify a compatibility of 2 or higher or do not specify a subfolder path."
```

Comment:
`tasks in subfolders are only supported version 2 and` Consider adding an `in` or an `on` between supported and version.